pkg/operator/ceph/cluster/mgr/spec.go

```diff
@@ -182,7 +184,13 @@ func (c *Cluster) makeMgrDaemonContainer(mgrConfig *mgrConfig) v1.Container {
 				Protocol:      v1.ProtocolTCP,
 			},
 		},
-		Env:       opspec.DaemonEnvVars(c.cephVersion.Image),
+		Env: append(
+			opspec.DaemonEnvVars(c.cephVersion.Image),
+			[]v1.EnvVar{
+				{Name: "ROOK_SYSTEM_NS", Value: operatorNamespace},
+				{Name: "ROOK_API_VERSION", Value: "v1"},
```

Comment:
Yup its the Ceph cluster CRD version.
`ROOK_API_VERSION` is hardcoded here https://github.com/ceph/ceph/blob/7ba6bece4187eda5d05a9b84211fe6ba8dd287bd/src/pybind/mgr/rook/rook_cluster.py#L29
so instead of hardcoding in rook orchestrator in ceph-mgr, we want to pass it to the container via Rook.

Reply:
May API version changes introduce API incompatibilities? 

If yes, we might want to revisit removing this hard-coded constant, as it is right now a good indicator of making sure the Rook version is compatible to the mgr module version.

Reply:
@sebastian-philipp @BlaineEXE @rohan47 What about using https://github.com/rook/rook/blob/master/pkg/apis/ceph.rook.io/v1/register.go#L28 ?

Reply:
Actually, after rethinking this, it would make sense to have the API version as an environment variable, in order to improve the orchestrator's error message, like

```python
if getenv('ROOK_API_VERSION') != ROOK_API_VERSION:
    return f"This Rook version is not supported".
```

Reply:
@galexrt using https://github.com/rook/rook/blob/master/pkg/apis/ceph.rook.io/v1/register.go#L28 sounds good to me

Reply:
@sebastian-philipp 
> May API version changes introduce API incompatibilities?
> 
Not sure but I think yes

> If yes, we might want to revisit removing this hard-coded constant, as it is right now a good indicator of making sure the Rook version is compatible to the mgr module version.
> 
https://github.com/rook/rook/blob/master/pkg/apis/ceph.rook.io/v1/register.go#L28 can be used

> We've discussed this topic in today's orchestrator meeting and it turned out that we have to make sure, either by version checks (like `ROOK_API_VERSION`) or by integration tests, that we catch API changes.

We can have an issue tracking this

Reply:
Perhaps we can include this as part of this helper method `ClusterDaemonEnvVars`?
Also I think @galexrt as a good point, we can use that as a reference.

Reply:
I just don't know how

> https://github.com/rook/rook/blob/master/pkg/apis/ceph.rook.io/v1/register.go#L28

could help.

Reply:
Versions are likely to be in the form k8s uses. e.g., v1, v2alpha1, v2beta1, v2beta2, v2, etc.

Reply:
And I would rather not put these in `ClusterDaemonEnvVars` since they are only used by the manager, and I don't see them being useful to other daemons.

Reply:
A few comments to catch up on this thread...
- Agreed that these new vars only belong on the mgr pod spec instead of adding it in `ClusterDaemonEnvVars`
- The version env var being proposed is the crd version. Today this should be `v1` and nothing else is supported. However, this doesn't give us a guarantee that the mgr module will be compatible with the crd version. This level of versioning hasn't been flushed out yet. The mgr module will need to decide what versions of the crd it supports. If the mgr module doesn't support the crd version passed in the env var, the module should likely fail. But if there is some finer-grained versioning needed, the mgr module will likely need to understand rook versioning. For example, rook could pass its version number (`v0.9.3`) as an env var.
- Let's add two more env vars: 
   - `ROOK_VERSION`: for versioning evaluation (as described in the previous bullet)
   - `ROOK_CLUSTER_NAME`: The `name` field of the cluster crd. The mgr module shouldn't need to guess the name of the crd (it might be different from the namespace, although not common)

Reply:
@travisn I updated this PR